disco/disco.go

```diff
@@ -29,14 +31,22 @@ const (
 	maxRedirects = 3
 
 	// Arbitrary-but-small time limit to prevent UI "hangs" during discovery.
-	discoTimeout = 11 * time.Second
+	defaultDiscoveryTimeout = 11 * time.Second
+
+	// Arbitrary number of times to retry discovery
+	defaultDiscoveryRetry = 5
 
 	// 1MB - to prevent abusive services from using loads of our memory.
 	maxDiscoDocBytes = 1 * 1024 * 1024
 )
 
-// httpTransport is overridden during tests, to skip TLS verification.
-var httpTransport = defaultHttpTransport()
```

Comment:
After your initial changes, the `defaultHttpTransport` function (defined in `http_transport.go`) is no longer called by anything. 

You could remove the function definition... or, alternatively, you could move the retryable HTTP transport setup into that function.

Speaking of that function, it looks like you'll probably want to re-create some of the setup that gets done for the struct it returns, especially setting up a default UserAgent.